bpdm-pool-api/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/api/PoolSiteApi.kt

```diff
@@ -133,4 +134,20 @@ interface PoolSiteApi {
         @ParameterObject searchRequest: SiteSearchRequest,
         @ParameterObject paginationRequest: PaginationRequest
     ): PageDto<SiteWithMainAddressVerboseDto>
+
+    @Operation(
+        summary = "Get page of sites matching the pagination search criteria",
+        description = "This endpoint retrieves all existing business partners of type sites."
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Page of business partners matching the search criteria, may be empty"),
+            ApiResponse(responseCode = "400", description = "On malformed pagination request", content = [Content()])
+        ]
+    )
+    @Tag(name = ApiCommons.SITE_NAME, description = ApiCommons.SITE_DESCRIPTION)
+    @PostMapping("/legal/reference")
```

Comment:
I'm not very happy with the resource name. Every site has a reference to a legal entity, which this name implies.

I propose /v6/legal-address-sites
or /v6/legal-main-sites

I'm open for suggestions but for me the name should be a resource name describing clearly a site that has a legal address as its main address, since this is the differentiating factor

Reply:
@nicoprow 

I added this logic to the PoolSiteApi, so /sites will follow v6. The endpoint will look like **/v6/sites/legal-address**.
Does it works?

Reply:
To be honest I'm not really happy with that path because it implies that you want to create a legal address that belongs to a site. But in fact, you want to create a site with a special property - that is - it has the legal address as its main address.

I'm open to suggestions but from my perspective the very last name in the path should describe the name of the resource you want to create which will be a site with the special property of having the legal entity's legal address as its site main address.

Reply:
As per our discussion, i was update the endpoint like **/v6/sites/legal-main-sites**.